product_variant_available_in_pos/models/product_template.py

```diff
@@ -0,0 +1,29 @@
+# Copyright 2016 Alex Comba - Agile Business Group
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, models
+
+
+class ProductTemplate(models.Model):
+
+    _inherit = 'product.template'
+
+    def _update_available_in_pos(self, vals):
+        values = {'available_in_pos': vals['available_in_pos']}
+        self.product_variant_ids.write(values)
+
+    @api.multi
+    def write(self, vals):
+        res = super(ProductTemplate, self).write(vals)
+        if 'available_in_pos' in vals:
+            for product in self:
+                product._update_available_in_pos(vals)
+        return res
+
+    @api.model
+    def create(self, vals):
```

Comment:
Are you sure this is needed? I think `product.product` `create` method already takes care of it.